### ISSUE TITLE
`ServeDir` does not redirect correctly for directories without trailing `/` when nested

### ISSUE THREAD

**Issue by Pistonight** [tower-rs/tower-http#413]
### Version
`tower-http v0.4.4`

### Platform
All

### Description
This function redirects paths that don't end with `/` and is a directory to the URI with `/` appended
https://github.com/tower-rs/tower-http/blob/e8eb54966604ea7fa574a2a25e55232f5cfe675b/tower-http/src/services/fs/serve_dir/open_file.rs#L253

However it does not take into account where the service could be nested, therefore producing an invalid redirect.

Example directory:
```
src/
├ main.rs
└ hello/
  └ index.html
```
main.rs:
```rust
use axum::{Router, Server};
use tower_http::services::ServeDir;

#[tokio::main]
async fn main() {
    tracing_subscriber::fmt::init();

    let serve_dir = ServeDir::new("src");
    let router = Router::new().nest_service("/src", serve_dir);

    let addr = "0.0.0.0:3000".parse().unwrap();

    Server::bind(&addr)
        .serve(router.into_make_service())
        .await
        .unwrap();

}

```

Going to `localhost:3000/src/hello` to see the issue
Expected: Redirects to `localhost:3000/src/hello/` and returns content at `src/hello/index.html`
Actual: Redirects to `localhost:3000/hello/` and returns Not Found

**Comment by jplatte** [comment]
Nesting is a concept of axum, not tower / tower-http. This is tracked at https://github.com/tokio-rs/axum/issues/1731.

**Comment by jplatte** [comment]
Yeah, I've thought a bunch about this in the meantime and I think it's probably much easier (and useful beyond axum) to add some prefix to `ServeDir` (maybe optionally stripping it from the input path, depending on whether you're already using sth. like nesting that does it up-front).

**Comment by maxi0604** [comment]
I have created a PR: #486
